test/test_integration.py

```diff
@@ -6,13 +6,12 @@
 import pytest
 
 from opensquirrel.circuit import Circuit
-from opensquirrel.default_instructions import CNOT, CZ, H
 from opensquirrel.ir import Measure
-from opensquirrel.passes.decomposer import CNOTDecomposer, McKayDecomposer, XYXDecomposer
+from opensquirrel.passes.decomposer import CNOT2CZDecomposer, CNOTDecomposer, McKayDecomposer, XYXDecomposer
 from opensquirrel.passes.exporter.export_format import ExportFormat
 
 
-def test_Spin2_backend() -> None:  # noqa: N802
+def test_spin_backend() -> None:
```

Comment:
Do we want to go from `Spin2` to `spin`?

```suggestion
def test_spin_2_backend() -> None:
```

Reply:
I left out the 2 on purpose; this integration test is valid for the spin system in general (the number of qubits is irrelevant). So also valid for the Spin-6 and later the Spin-8 system.

Reply:
Great!